### ISSUE TITLE
Verbalize current page of search results pagination

### ISSUE THREAD

**Issue by atsansone** [flutter/website#10539]
From https://buganizer.corp.google.com/issues/337306005

### Observed Results

Navigating through the page reaching the pagination, listen that the current/selected page is not announced by SR.

* [Screencast VoiceOver](http://screencast/cast/NDgyNjYyMzMyMzkzMDYyNHwyNTc1MjA3Mi1mZQ)
* [Screencast ChromeVox](http://screencast/cast/NjIzNTE4NzE1MDA2MTU2OHw0MmM0YzM1MC0wOQ)

### Expected Results

Ensure that when the user is navigating through the pagination, the SR should announce what is the current selected page which is open.

### User Impact

Screen reader and Cognitive users are affected because they are not aware what the current page is when they are navigating through the pagination.

### Steps to Reproduce

1. Enabled VoiceOver using <kbd>Command</kbd> + <kbd>F5</kbd> also ChromeVox using "Ctrl+Alt+Z'
1. Open the following page: https://docs.flutter.dev/search?q=devtools
1. Continue the linear navigation using the <kbd>Tab</kbd> key.
1. Reach the pagination component and listen how the current page is announced.

### GAR Assessment

[Flutter Docs](https://a11ytracker.corp.google.com/products/c2fdcf18-53d2-475d-b8b0-e5dedf8cb900?e=ProductLabels)

### GAR Details

[1.1. Do all non-text content and user interface components have meaningful accessible names and alternative text?](go/GAR-web#accessible-names)

### Environment Details

Machine: MAC/ChromeBook
Browser: Chrome Version 120.0.6099.272 (Official Build) (64-bit)
Screen reader: VoiceOver / ChromeVox.
Bug filling time: 15 mins.

**Comment by parlough** [comment]
These search results are generated, styled, and managed by the Google Custom Search setup we use.

Since we don't control that, I'm going to close this in favor of an umbrella issue https://github.com/flutter/website/issues/11800 which tracks implementing a new experience that we can properly integrate and style. Thanks!
